zilliqa/src/consensus.rs

```diff
@@ -1582,7 +1577,7 @@ impl Consensus {
         }
 
         // Derive the proposer from the block's view
-        let proposer = self.leader_at_block(&parent, block.view()).unwrap();
+        let proposer = self.leader_at_block(&parent).unwrap();
```

Comment:
We should use the `leaderAtView()` function of the `Deposit` contract and pass in the `view` argument, otherwise if we execute `leader_at_block()` on the `parent` the view number that determines the randomness in the `Deposit::leader()` function will be the `parent`'s view and not `block.view()`.